internal/utils/decoders/decoders.go

```diff
@@ -35,55 +32,76 @@ func NormalizeKeys(keys map[string]string, rootName string, sep string) map[stri
 			continue
 		}
 
-		if strings.TrimSpace(strings.TrimSuffix(strings.TrimPrefix(lowerKey, "providers"+sep), strings.ReplaceAll(suffix, "-", sep))) == "" {
+		parts = append(parts, root)
+
+		id := strings.TrimPrefix(key, root+"-")
+		id = strings.TrimSuffix(id, "-"+suffix)
+
+		if id == "" {
 			continue
 		}
 
-		clientNameParts := strings.Split(strings.TrimPrefix(strings.TrimSuffix(lowerKey, sep+strings.ReplaceAll(suffix, "-", sep)), "providers"+sep), sep)
+		parts = append(parts, id)
+		parts = append(parts, suffix)
+
+		final := ""
 
-		for i, p := range clientNameParts {
+		for i, part := range parts {
 			if i == 0 {
-				camelClientName += p
+				final += kebabToCamel(part)
 				continue
 			}
-			if p == "" {
-				continue
-			}
-			camelClientName += strings.ToUpper(string([]rune(p)[0])) + string([]rune(p)[1:])
+			final += "." + kebabToCamel(part)
```

Comment:
This is appending strings in two lines with almost duplicate code, slightly cleaner might be -

```suggestion
			if i > 0 {
				final += "."
			}
			final += kebabToCamel(part)
```